include/alpaka/rand/RandAlpaka.hpp

```diff
@@ -0,0 +1,200 @@
+/* Copyright 2021 Jeffrey Kelling
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Unused.hpp>
+#include <alpaka/math/cos/Traits.hpp>
+#include <alpaka/math/log/Traits.hpp>
+#include <alpaka/math/sin/Traits.hpp>
+#include <alpaka/math/sqrt/Traits.hpp>
+#include <alpaka/rand/RandStdLib.hpp>
+#include <alpaka/rand/Traits.hpp>
+
+#include <limits>
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace rand
+    {
+        //! "Tiny" state mersenne twister implementation
+        class TinyMersenneTwisterGpu : public concepts::Implements<ConceptRand, TinyMersenneTwisterGpu>
+        {
+        };
+        using RandAlpaka = TinyMersenneTwisterGpu;
```

Comment:
I don't like the name `RandAlpaka`. Please either call it `RandTinyMersenneTwister` to indicate what implemenatation it is, or if that does not matter we could call it `RandDefault`. But I prefer the first one.

This would also affect the header names unfortunately.

Reply:
As #1319 is merged now: @sliwowitz do you think it would be simple to use the new Philox instead of the TinyMT?

We can rename to RandDefault.

Reply:
I think it should be possible. I'll give it a try and report back here.

Reply:
Please check the register-footprint of `Philox` before making it the default. As I know from @sliwowitz `Philox` is generating 4 numbers within one call so it could be generating rng number that could have a larger footprint.
A large memory footprint means mostly slower kernel execution.

Reply:
It is a fallback, not yet supposed to be the default for all `Acc`s. Using it as default for other backends should be discussed separately.